src/libraries/System.IO.FileSystem/tests/File/OpenHandle.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32.SafeHandles;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    // to avoid a lot of code duplication, we reuse FileStream tests
+    public class File_OpenHandle : FileStream_ctor_options_as
+    {
+        protected override string GetExpectedParamName(string paramName) => paramName;
+
+        protected override FileStream CreateFileStream(string path, FileMode mode)
+        {
+            FileAccess access = mode == FileMode.Append ? FileAccess.Write : FileAccess.ReadWrite;
+            return new FileStream(File.OpenHandle(path, mode, access, preallocationSize: PreallocationSize), access);
+        }
+
+        protected override FileStream CreateFileStream(string path, FileMode mode, FileAccess access)
+            => new FileStream(File.OpenHandle(path, mode, access, preallocationSize: PreallocationSize), access);
+
+        protected override FileStream CreateFileStream(string path, FileMode mode, FileAccess access, FileShare share, int bufferSize, FileOptions options)
+            => new FileStream(File.OpenHandle(path, mode, access, share, options, PreallocationSize), access, bufferSize, (options & FileOptions.Asynchronous) != 0);
+
+        [Fact]
+        public override void NegativePreallocationSizeThrows()
+        {
+            ArgumentOutOfRangeException ex = Assert.Throws<ArgumentOutOfRangeException>(
+                () => File.OpenHandle("validPath", FileMode.CreateNew, FileAccess.Write, FileShare.None, FileOptions.None, preallocationSize: -1));
+        }
+
+        [Theory, MemberData(nameof(StreamSpecifiers))]
+        public override void FileModeAppendExisting(string streamSpecifier)
+        {
+            // currently not enabled due to https://github.com/dotnet/runtime/issues/53432
```

Comment:
If we open a handle for `FileMode.Append`, there is no `FileStream` ctor that accepts a handle and `FileMode`. So append test that receive `FileStream` created from such a handle fail.

Moreover, we don't use native flags like `O_APPEND` mentioned in #53432 so if user provides us just a handle, we can't always detect whether it was opened for append or not (I plan to change that very soon as this would increase reliability as OS would ensure that the content is always appended).